tests/app/main/test_suppliers.py

```diff
@@ -2246,3 +2246,76 @@ def test_existing_org_size_sets_current_selection(self, data_api_client, existin
             doc = html.fromstring(res.get_data(as_text=True))
             selected_value = doc.xpath('//input[@name="organisation_size" and @checked="checked"]/@value')
             assert selected_value == expected_selection, 'The organisation size has not pre-populated correctly.'
+
+
+@mock.patch("app.main.views.suppliers.data_api_client", autospec=True)
+class TestSupplierAddRegisteredCompanyName(BaseApplicationTest):
+    def test_add_registered_company_name_page_loads(self, data_api_client):
+        with self.app.test_client():
+            self.login()
+            data_api_client.get_supplier.return_value = get_supplier(registeredName=None)
+
+            res = self.client.get("/suppliers/registered-company-name/edit")
+            assert res.status_code == 200, 'The add registered company name page has not loaded correctly.'
+
+    def test_no_input_triggers_input_required_validation_and_does_not_call_api_update(self, data_api_client):
+        with self.app.test_client():
+            expect_header = self.strip_all_whitespace('There was a problem with your answer to:Registered company name')
+            expect_error = 'You must provide a registered company name.'
```

Comment:
Can we use `self.assert_validation_masthead` here? Maybe it could be updated to also take a single question validation error to check for elsewhere to encapsulate some more of this check.

Reply:
I can try it.